src/server/session.ts

```diff
@@ -667,6 +703,15 @@ namespace ts.server {
             this.projectService = new ProjectService(settings);
             this.projectService.setPerformanceEventHandler(this.performanceEventHandler.bind(this));
             this.gcTimer = new GcTimer(this.host, /*delay*/ 7000, this.logger);
+
+            // Make sure to setup handlers to throw error for not allowed commands on syntax server;
+            if (this.projectService.syntaxOnly) {
+                invalidSyntaxOnlyCommands.forEach(commandName =>
+                    this.handlers.set(commandName, request => {
+                        throw new Error(`Request: ${request.command} not allowed on syntaxServer:: Request::${JSON.stringify(request)}`);

```

Comment:
Why do we need the request?  In the log, it doesn't matter if it's well-formed - it's that it got sent at all.  It'll be immediately before the error in the log anyway, right?

Reply:
Do we want to test this rejection mechanism?  Not for all commands, but maybe for one?

Reply:
Already there https://github.com/microsoft/TypeScript/pull/38561/files#diff-0403b837aaec2ba7b3953efef687f19aR66 tests both LS and session.
This is to ensure that plugins or anyone doesnt accidently use semantic operation on LS or session from API point of view